packages/esm-form-entry-app/src/app/form-data-source/form-data-source.service.ts

```diff
@@ -114,6 +117,56 @@ export class FormDataSourceService {
     );
   }
 
+  public getMostRecentObsValueBefore(date: string, patientUuid: string) {
+    type ObsResult = FHIRResource['resource'] & {
+      conceptUuid: string;
+      dataType?: string;
+    };
+
+    function isUuid(input: string) {
```

Comment:
It's a bad idea to declare a function inside a function, especially as in this case where there's to dependency on data in the containing function. The reason is that functions declared inside a function are re-created on every function execution, i.e., we're asking the interpreter to do unnecessary work.

Reply:
oh, I get it, I will move it out. Thanks for that tip.